Python/Product/TestAdapter.Executor/Pytest/TestResultParser.cs

```diff
@@ -23,69 +23,61 @@
             }
+            return null;
+        }
 
-            var doc = Read(junitXmlPath);
-            XPathNodeIterator xmlTestCases = doc.CreateNavigator().Select("/testsuite/testcase");
-
-            foreach (XPathNavigator pytestcase in xmlTestCases) {
-                var file = pytestcase.GetAttribute("file", "");
-                var name = pytestcase.GetAttribute("name", "");
-                var classname = pytestcase.GetAttribute("classname", "");
-
-                if (String.IsNullOrEmpty(file) ||
-                    String.IsNullOrEmpty(name) ||
-                    String.IsNullOrEmpty(classname) ||
-                    !Int32.TryParse(pytestcase.GetAttribute("line", String.Empty), out int line))
-                {
-                    var message = String.Empty;
-                    if(pytestcase.HasChildren) {
-                        pytestcase.MoveToFirstChild();
-                        message = pytestcase.GetAttribute("message", String.Empty) + pytestcase.Value;
-                    }
-                    Debug.WriteLine("Test result parse failed: {0}".FormatInvariant(message));
-                    continue;
-                }
-
-                // Match on classname and function name for now
-                var result = vsTestResults
-                    .Where(tr =>
-                    String.Compare(tr.TestCase.DisplayName, name, StringComparison.InvariantCultureIgnoreCase) == 0 &&
-                    String.Compare(tr.TestCase.GetPropertyValue<string>(Pytest.Constants.PyTestXmlClassNameProperty, default(string)), classname, StringComparison.InvariantCultureIgnoreCase) == 0) 
-                    .FirstOrDefault();
-
-                if (result != null) {
-                    UpdateTestResult(pytestcase, result);
-                } else {
-                    var message = String.Empty;
-                    if (pytestcase.HasChildren) {
-                        pytestcase.MoveToFirstChild();
-                        message = pytestcase.GetAttribute("message", String.Empty) + pytestcase.Value;
-                    }
-                    Debug.WriteLine("Testcase for result not found: {0}".FormatInvariant(message));
-                }
+        /// <summary>
+        /// Handles two cases:
+        /// case A: Function instead a class
+        ///  <testcase classname="test2.Test_test2" file="test2.py" name="test_A" >
+        ///  pytestId .\test2.py::Test_test2::test_A
+        ///  
+        /// case B: Global function
+        ///   <testcase classname="test_sample" file="test_sample.py" name="test_answer" >
+        ///   pytestId .\test_sample.py::test_answer 
+        /// </summary>
+        /// <param name="filename"></param>
+        /// <param name="classname"></param>
+        /// <param name="funcname"></param>
+        /// <returns></returns>
+        internal static string CreatePytestId(string filename, string classname, string funcname) {
+            var classParts = classname.Split('.');
```

Comment:
This seems to be splitting only to find the first `'.'`, then merging everything after it, so a bit of a waste of memory allocations.

Should be simple with just `IndexOf`
```
var classIndex = classname.IndexOf('.');
if (classIndex >= 0 && classIndex < (classname.Length - 1)) {
    var classNameWithoutFileName = classname.Substring(classIndex + 1);
    ...
} else {
   ...
}

Reply:
cool. ya substring will work, i'll make this change

Reply:
updated